### ISSUE TITLE
syntax checking

### ISSUE THREAD

**Issue by lwaldron** [waldronlab/curatedMetagenomicData#10]
Perform syntax checking using `inst/extdata/metadata/checkCurated.R` and `inst/extdata/metadata/template_metagenomic.csv`. Also add the template to featureData slot of the ExpressionSets.

**Comment by schifferl** [comment]
With the refactoring of the repository, these syntax checks are no longer relevant. I am closing this issue but could write new syntax checks if you think they are needed.

**Comment by lwaldron** [comment]
What do you mean they're not relevant? Do you mean they are impractical as
unit tests? How will you ensure that a manually curated phenoData variable
that is supposed to be numeric or integer doesn't contain a "O" instead of
a "0" somewhere, or that a disease state that is supposed to be "cancer"
isn't "Cancer" or "cacner". These kinds of curation mistakes happen, and
the template with curatedOvarianData syntax checker will catch them as long
as you apply them somewhere in the pipeline.

**Comment by schifferl** [comment]
Ok, I didn't understand correctly then. I will reopen the issue and incorporate the script into the pipeline.

**Comment by lwaldron** [comment]
Here is an example of the kind of problem that syntax checking will avoid. `subjectID` would be required to be of class character, even if it contains only integer values:

```
> library(dplyr)
> full_join(pData(eset.list[[1]]), pData(eset.list[[2]]))
Joining, by = c("subjectID", "bodysite", "disease", "country", "age", "number_reads")
Error in eval(substitute(expr), envir, enclos) : 
  Can't join on 'subjectID' x 'subjectID' because of incompatible types (character / integer)
> names(eset.list)[1:2]
[1] "HMP_2012.metaphlan_bugs_list.stool"       
[2] "KarlssonFH_2013.metaphlan_bugs_list.stool"> 

```

**Comment by lwaldron** [comment]
Depends on #38

**Comment by lwaldron** [comment]
Following rows of template.csv are too lenient:
```
bodysite,character,non-unique,required,*,Bodysite of acquisition
disease,character,non-unique,required,*,Disease presence and type
age,numeric,non-unique,required,*,Subject age (years)
```

**Comment by lwaldron** [comment]
Added `checkCuration()` function and found a lot of curation issues in `combined_metadata`, see attached Excel spreadsheet.

[curationproblems.xlsx](https://github.com/waldronlab/curatedMetagenomicData/files/981277/curationproblems.xlsx)

**Comment by edoardopasolli** [comment]
I think this issue is closed.
template.csv file was re-done during the rebuilding of the package
all metadata files went successfully through the checkCuration() function
